src/firm_ce/constructors/topology_cons.py

```diff
@@ -322,9 +329,9 @@ def construct_Network_object(
     """
 
     nodes = TypedDict.empty(key_type=int64, value_type=Node_InstanceType)
-    for idx in range(len(nodes_imported_list)):
+    for idx in range(len(nodes_imported_dict)):
```

Comment:
nodes_imported_dict is keyed by Node.order. This line should just be 'for idx in nodes_imported_dict'.